tests/integrationtest/t/executor/grant.test

```diff
@@ -0,0 +1,289 @@
+# TestWithGrantOption
+drop user if exists 'testWithGrant'@'localhost';
+CREATE USER 'testWithGrant'@'localhost' IDENTIFIED BY '123';
+SELECT * FROM mysql.db WHERE User="testWithGrant" and host="localhost";
+GRANT select ON executor__grant.* TO 'testWithGrant'@'localhost' WITH GRANT OPTION;
+SELECT grant_priv FROM mysql.DB WHERE User="testWithGrant" and host="localhost" and db="executor__grant";
+drop user if exists 'testWithGrant1';
+CREATE USER 'testWithGrant1';
+SELECT grant_priv FROM mysql.user WHERE User="testWithGrant1";
+GRANT ALL ON *.* TO 'testWithGrant1';
+SELECT grant_priv FROM mysql.user WHERE User="testWithGrant1";
+GRANT ALL ON *.* TO 'testWithGrant1' WITH GRANT OPTION;
+SELECT grant_priv FROM mysql.user WHERE User="testWithGrant1";
+
+# TestIssue2456
+drop user if exists 'dduser'@'%';
+drop DATABASE if exists `dddb_%`;
+CREATE USER 'dduser'@'%' IDENTIFIED by '123456';
+CREATE DATABASE `dddb_%`;
+CREATE table `dddb_%`.`te%` (id int);
+GRANT ALL PRIVILEGES ON `dddb_%`.* TO 'dduser'@'%';
+GRANT ALL PRIVILEGES ON `dddb_%`.`te%` to 'dduser'@'%';
+
+# TestNoAutoCreateUser
+DROP USER IF EXISTS 'test'@'%';
+SET sql_mode='NO_AUTO_CREATE_USER';
+-- error 1410
+GRANT ALL PRIVILEGES ON *.* to 'test'@'%' IDENTIFIED BY 'xxx';
+set sql_mode=default;
+
+# TestCreateUserWhenGrant
+DROP USER IF EXISTS 'test'@'%';
+# This only applies to sql_mode:NO_AUTO_CREATE_USER off
+SET SQL_MODE='';
+GRANT ALL PRIVILEGES ON *.* to 'test'@'%' IDENTIFIED BY 'xxx';
+# Make sure user is created automatically when grant to a non-exists one.
+SELECT user FROM mysql.user WHERE user='test' and host='%';
+DROP USER IF EXISTS 'test'@'%';
+# Grant without a password.
+GRANT ALL PRIVILEGES ON *.* to 'test'@'%';
+# Make sure user is created automatically when grant to a non-exists one.
+SELECT user, plugin FROM mysql.user WHERE user='test' and host='%';
+DROP USER IF EXISTS 'test'@'%';
+set sql_mode=default;
+
+# TestCreateUserWithTooLongName
+-- error 1470
```

Comment:
Do we need to drop user first? Will this test be run repeatedly?

Reply:
Drop user will report error, the user name is too long.

Reply:
ok